src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
it's better to set full variable name: person, man

```suggestion
const children = onlyWithSon
  ? people.filter(person=> person.sex === 'm' && people.some(man => man.name === person.mother))
  : people.filter(person => people.some(man => man.name === person.mother));
```